qadence_protocols/mitigations/twirl_mitigation.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+
+import itertools
+from functools import reduce
+
+import torch
+
+# import pytest
+from qadence import (
+    QuantumCircuit,
+    QuantumModel,
+    chain,
+)
+from qadence.noise.protocols import Noise
+from qadence.operations import X, kron
+from qadence.types import BackendName
+
+# from metrics import MIDDLE_ACCEPTANCE
+
+
+def twirl_swap(n_qubits: int, twirl: tuple, samples_twirl: dict) -> dict:
+    output = {}
+    operand = "".join(map(str, [1 if i in twirl else 0 for i in range(n_qubits)]))
+    for key in samples_twirl.keys():
+        output["{0:b}".format(int(key, 2) ^ int(operand, 2))] = samples_twirl[key]
+
+    return output
+
+
+def compute_exp(n_qubits: int, samples_twirl_list: list, observable: list) -> float:
+    out: float = 0
+    Z_mat = torch.eye(2)
+    Z_mat[1, 1] = -1
+    o = reduce(torch.kron, [Z_mat if i in observable else torch.eye(2) for i in range(n_qubits)])
+
+    for samples_twirl in samples_twirl_list:
```

Comment:
I would sungularise here.

```suggestion
    for sample_twirl in samples_twirl:
```